app/modules/individuals/models.py

```diff
@@ -231,15 +287,129 @@ def merge_from(self, *source_individuals, sex=None, primary_name=None):
+                owners[owner]['encounters'],
+                request_data,
+                notif_type,
+            )
+
+    @classmethod
+    def _merge_request_notify_user(
+        cls, sender, user, individuals, encounters, request_data, notif_type
+    ):
+        from app.modules.notifications.models import (
+            Notification,
+            NotificationBuilder,
+        )
+
+        builder = NotificationBuilder(sender)
+        builder.set_individual_merge_request(individuals, encounters, request_data)
+        notification = Notification.create(notif_type, user, builder)
+        log_msg = (
+            f'merge request: notification {notification} from {sender} re: {individuals}'
+        )
+        AuditLog.audit_log_object(log, user, log_msg)
+
+    @classmethod
+    def merge_request_cancel_task(cls, req_id):
+        current_app.celery.control.revoke(req_id)
+        Individual.merge_request_cleanup(req_id)
+
+    # scrubs notifications etc, once a merge has completed
+    @classmethod
+    def merge_request_cleanup(cls, req_id):
+        log.debug(
+            f'[{req_id}] merge_request_cleanup (notifications, etc) not yet implemented'
+        )
+
+    @classmethod
+    def merge_request_celery_task(
+        cls, cel_task, target_individual_guid, from_individual_ids, parameters
+    ):
+        from app.modules.notifications.models import NotificationType
+
+        log_id = f'<execute_merge_request {cel_task.request.id}>'
+        log.info(
+            f'{log_id} initiated for Individual {target_individual_guid} (from {from_individual_ids}; {parameters})'
+        )
+        all_individuals = Individual.validate_merge_request(
+            target_individual_guid, from_individual_ids, parameters
+        )
+        if not all_individuals:
+            msg = f'{log_id} failed validation'
+            AuditLog.backend_fault(log, msg)
+            return
+
+        # validate_merge_request should check hashes etc and means we are good to merge
+        target_individual = all_individuals.pop(0)
+        # TODO additional work TBD regarding conflict args/parameters (DEX-514)
+        try:
+            res = target_individual.merge_from(*all_individuals)
+        except Exception as ex:
+            res = f'Exception caught: {str(ex)}'
+        if not isinstance(res, dict):
+            msg = f'{log_id} (via celery task) merge_from failed: {res}'
+            AuditLog.backend_fault(log, msg)
+            return
+
+        log.info(f'{log_id} merge completed, results={res}')
+
+        # notify users that merge has happened
+        #   NOTE request_data here may need some altering depending on what final templates look like
+        #   also unclear who *sender* will be, so that may need to be passed
+        request_data = {
+            'id': cel_task.request.id,
+            'from_individual_ids': from_individual_ids,
+            'merge_outcome': 'deadline',
+        }
+        Individual.merge_request_notify(
```

Comment:
surely this is a notification of the merge itslef, not the request? I'd say that the word request here is misleading